aptos-move/framework/aptos-token-objects/sources/token.move

```diff
@@ -77,25 +97,83 @@ module aptos_token_objects::token {
         creator_address: address,
         collection_name: String,
         description: String,
-        name: String,
+        name_prefix: String,
+        name_with_index_suffix: Option<String>,
```

Comment:
Can we make this a format string with 1 pair of `{}` for index?
This can give user more flexibility of the name format. Such as `aaron_token_{}`

Reply:
they can do that too:
name_prefix = "aaron_token_"
name_suffix= option::some("")

currently , I didn't do {}, because I didn't want to support varargs for now, but might want to in the future.

so we might add format() with multiple arguments in the future, and I want to keep the syntax for that time